components/ord-service/src/main/java/com/sap/cloud/cmp/ord/service/controller/SpecificationsController.java

```diff
@@ -23,16 +33,40 @@
     private EventSpecRepository eventSpecRepository;
 
 
-    @RequestMapping(value = "/${odata.jpa.request_mapping_path}/api/{id}/specification", method = { RequestMethod.GET }, produces = MediaType.TEXT_PLAIN_VALUE)
+    @RequestMapping(value = "/${odata.jpa.request_mapping_path}/api/{id}/specification", method = {RequestMethod.GET}, produces = MediaType.TEXT_PLAIN_VALUE)
     @ResponseBody
-    public String getApiSpec(@PathVariable final String id) {
-        return apiSpecRepository.getOne(UUID.fromString(id)).getSpecData();
+    public String getApiSpec(HttpServletRequest request, HttpServletResponse response, @PathVariable final String id) throws IOException {
+        APISpecificationEntity apiSpec = new APISpecificationEntity();
+        String tenantID = extractInternalTenantIdFromIDToken(request);
+
+        try {
+            apiSpec = apiSpecRepository.getByApiDefinitionIdAndTenant(UUID.fromString(id), UUID.fromString(tenantID));
+            if (apiSpec == null) {
+                response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+                return NOT_FOUND_MESSAGE;
+            }
+        } catch (IllegalArgumentException e) {
+            super.handleErrorResponse(response, INVALID_TENANT_ID_ERROR_MESSAGE, CONTENT_TYPE_PLAIN_TEXT);
+        }
```

Comment:
Is this try catch even needed? When `IlegalArgumentException` is thrown. Also if it is the case you invoke handleErrorResponse with plain text content type and the handleErrorResponse method writes JSON output.

Reply:
It is probably because of the `UUID.fromString(...)`

Still, we can change handlerErrorResponse to not return JSON when it is plain-text.

Also, this method handlerErrorResponse is only used in specifications controller, it does not make much sense to be in the base controller we can add it here and hardcode it to return the specific response for the specifications for simplicity.

Reply:
Yes, that's the reason why the `try catch` is needed.

We thought whether or not to leave the method in the base controller and we decided to leave it there because it has kind of generic functionality and can be useful in the future if another controller is introduced. 

Your suggestion is valid as well. What are your thoughts when comparing these two approaches?

Reply:
Discussed offline with @DimitarPetrov. We decided that it's better to move the `handleErrorResponse` method in this particular controller from the base one.

The reasoning is that this API is hardcoded to return plain text response body (can be seen in the method's annotation) and it doesn't make much sense to wrap the plain text message in a JSON structure.